torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -112,6 +112,59 @@ def replace_no_op(node, replace_input_index):
             replace_no_op(node, 0)
 
 
+@torch.utils._python_dispatch._disable_current_modes()
+def remove_redundant_views(gm: torch.fx.GraphModule):
+    """
+    Removes redundant views by reusing existing ones.
+    """
+
+    # A dictionary mapping a tensor to all aliased views.
+    views: Dict[torch.fx.Node, List[torch.fx.Node]] = {}
+    graph = gm.graph
+
+    for node in graph.nodes:
+        if node.op != "call_function":
+            continue
+
+        if node.target != torch.ops.aten.view.dtype:
+            continue
+
+        src = node.args[0]
+        to_type = node.args[1]
+        existing_views = views.get(src)
+        is_needed = True
+
+        if existing_views:
+            # Replace the view with the an existing view if available.
+            for alias in existing_views:
```

Comment:
Rather than using a loop here, you can modify the views dict key to be:
```
(source_node, from_dtype, to_dtype)
```

Reply:
Thanks for the suggestion. I thought about that actually, but given that `aten.view` is bidirectional, we actually constructing a acyclic graph for a given alias closure. I'm using a shared list to store all nodes in the closure. Alternatively as you suggested, we can store each graph edge in separately. 

I feel that the current implementation is simpler, and of course each search takes in theory linear time to finish. Do you think that is a concern? How big an alias closure can be in reality?

With the alternative, the search for a view with desired type can be as complex as a DFS or BFS, and we may need to handle cycles. Also edges are usually more than nodes, and I'd expect it causes more memory to build the dictinoary.

Reply:
Updated to using a dictionary instead of a list for efficiency.